pyuvdata/uvcal.py

```diff
@@ -583,3 +583,407 @@ def write_calfits(self, filename, run_check=True, check_extra=True,
                                   run_check_acceptability=run_check_acceptability,
                                   clobber=clobber)
         del(calfits_obj)
+
+    def __add__(self, other, run_check=True, check_extra=True,
+                run_check_acceptability=True, inplace=False):
+        """
+        Combine two UVCal objects. Objects can be added along antenna, frequency,
+        time, and/or Jones axis.
+
+        Args:
+            other: Another UVCal object which will be added to self.
+            run_check: Option to check for the existence and proper shapes of
+                parameters after combining objects. Default is True.
+            check_extra: Option to check optional parameters as well as
+                required ones. Default is True.
+            run_check_acceptability: Option to check acceptable range of the values of
+                parameters after combining objects. Default is True.
+            inplace: Overwrite self as we go, otherwise create a third object
+                as the sum of the two (default).
+        """
+        if inplace:
+            this = self
+        else:
+            this = copy.deepcopy(self)
+        # Check that both objects are UVCal and valid
+        this.check(check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+        if not isinstance(other, this.__class__):
+            raise(ValueError('Only UVCal objects can be added to a UVCal object'))
+        other.check(check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+
+        # Check objects are compatible
+        # Note zenith_ra will not necessarily be the same if times are different.
+        # But phase_center should be the same, even if in drift (empty parameters)
+        compatibility_params = ['_cal_type', '_integration_time', '_channel_width',
+                                '_telescope_name', '_gain_convention', '_x_orientation']
```

Comment:
Good idea. I don't want to be too restrictive, so I think a warning is appropriate.